CMakeLists.txt

```diff
@@ -191,9 +191,13 @@ if (CMAKE_C_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wno-gnu-statement-expression)
 endif()
 
-if (NOT (APPLE OR USE_SANITIZERS OR USE_TSAN))
+if (APPLE)
```

Comment:
Just a head's up - I'm refactoring the way we enable TSAN in cmake to follow the model used by proton. 
See [DISPATCH-1450](https://issues.apache.org/jira/browse/DISPATCH-1450)

I'll be creating a pull request for DISPATCH-1450 in the next day or so.

Reply:
Thanks for the head's up. Would you be OK if this gets merged before you get done? For APPLE, the important thing is never to set `"-Wl,-z,defs"`, because then the compile fails. No other considerations I know about, so far.

Btw, there is apparently special automagical setting to enable sanitizers with XCODE

```
cmake_minimum_required(VERSION 3.13)
set(CMAKE_XCODE_GENERATE_SCHEME ON)
set(CMAKE_XCODE_SCHEME_ADDRESS_SANITIZER ON)
set(CMAKE_XCODE_SCHEME_ADDRESS_SANITIZER_USE_AFTER_RETURN ON)
```

https://stackoverflow.com/questions/44320465/whats-the-proper-way-to-enable-addresssanitizer-in-cmake-that-works-in-xcode

Probably something that can be safely ignored for now.

I was wondering if "modern" CMake has special support for Valgrind, and so on, but I found nothing. It has e.g. support for cppcheck, though, https://cmake.org/cmake/help/v3.10/prop_tgt/LANG_CPPCHECK.html

Reply:
yep, np.  I'll hold off until this patch lands.

Reply:
I'll consider comment https://github.com/jdanekrh/qpid-dispatch/commit/4c07a4d9f5e0bf6c12a3d5661f86adc6390b432a#r35553494 as sufficient review and merge this.